components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/factory/RegistrationRequestFactory.java

```diff
@@ -114,46 +113,37 @@ public void create(IdentityRequest.IdentityRequestBuilder builder, HttpServletRe
                 registrationRequestProfile = new RegistrationRequestProfile();
             }
 
-            Object obj = jsonData.get(RegistrationRequest.RegisterRequestConstant.REDIRECT_URIS);
+            Object obj = jsonData.get(RegistrationRequest.RegisterRequestConstant.GRANT_TYPES);
             if (obj != null && obj instanceof JSONArray) {
-                JSONArray redirectUris = (JSONArray) obj;
-                for (Object redirectUri : redirectUris) {
-                    //ToDO invalid_redirect_uri - error code should be sent to client in below error cases.
-
-                    if (redirectUri == null || IdentityUtil.isBlank(redirectUri.toString())) {
-                        throw IdentityException
-                                .error(FrameworkClientException.class, "The redirection URI is either null or blank.");
+                JSONArray grantTypes = (JSONArray) obj;
+                for (Object grantType : grantTypes) {
+                    if (IdentityUtil.isNotBlank((String) grantType)) {
+                        registrationRequestProfile.getGrantTypes().add((String) grantType);
                     }
+                }
+            } else if (IdentityUtil.isNotBlank((String) obj)) {
+                registrationRequestProfile.getGrantTypes().add((String) obj);
+            }
 
-                    String redirectUriStr = redirectUri.toString();
-                    try {
-                        //Trying to parse the URI, just to verify the URI syntax is correct.
-                        URI redirectURL = new URI(redirectUriStr);
-                    } catch (URISyntaxException e) {
-                        String errorMessage = "The redirection URI " + redirectUriStr + " is not a valid URI.";
-                        throw IdentityException.error(FrameworkClientException.class, errorMessage, e);
-                    }
-                    registrationRequestProfile.getRedirectUris().add(redirectUriStr);
+            obj = jsonData.get(RegistrationRequest.RegisterRequestConstant.REDIRECT_URIS);
+            if (obj != null && obj instanceof JSONArray) {
```

Comment:
We do not need to check (obj != null) when (obj instanceof X) is checked.
http://docs.oracle.com/javase/specs/jls/se7/html/jls-15.html#jls-15.20.2